core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/delete/KeystoreFileManager.java

```diff
@@ -55,6 +60,39 @@ public void deleteKeystoreFiles(final String pubkey) throws IOException {
     }
   }
 
+  /**
+   * Create Keystore metadata, json and password files
+   *
+   * @param fileNameWithoutExtension File name, usually public key, without extension.
+   * @param jsonKeystoreData Keystore Json data which will be written to
+   *     fileNameWithoutExtension.json
+   * @param password password char[] which will be written as pubkey.password. All the values in the
+   *     array will be reset after password is written.
+   * @throws IOException In case file write operations fail
+   */
+  public void createKeystoreFiles(
+      final String fileNameWithoutExtension, final String jsonKeystoreData, final char[] password)
+      throws IOException {
+    final Path metadataYamlFile = keystorePath.resolve(fileNameWithoutExtension + YAML_EXTENSION);
+    final Path keystoreJsonFile = keystorePath.resolve(fileNameWithoutExtension + JSON_EXTENSION);
+    final Path keystorePasswordFile =
+        keystorePath.resolve(fileNameWithoutExtension + PASSWORD_EXTENSION);
+
+    // create yaml file first (so that if it fails we haven't written password file before it)
```

Comment:
Do you think we also need to cleanup the other files in `removeSignersAndCleanupImportedKeystoreFiles`?
Or at least the keystore json file if e.g. writing the password file fails.

I know it's not directly related to your changes, but your comment made me realise there was actually three files to cleanup not just the yaml file.

Reply:
Also really like that this comment explains the _why_ :)